R/give_opinions.R

```diff
@@ -1,127 +1,169 @@
+
+  # read description from description_file if description is NULL
+  description <- read_description_if_null(description, description_file)
+
+  # Start with an empty vector of "fixme" messages
+  fixmes <- character()
+
   # Authors
-  if (inherits( try(descr$get_authors(), silent = TRUE),
-                'try-error')){
-    authors_fixme <- "The syntax Authors@R instead of plain Author and Maintainer is recommended in particular because it allows richer metadata" # no lint
-  }else{
-    authors_fixme <- NULL
+  if (fails(description$get_authors())) {
+
+    fixmes <- c(fixmes, get_message(message_id = "hint_use_authors_r"))
   }
 
   # URL
-  if(is.na(descr$get("URL"))){
-    url_fixme <- "URL field. Indicate the URL to your code repository."
-  }else{
-    checkurls <- check_urls(descr$descr$get_urls)
-    if(checkurls != ""){
-      url_fixme <- checkurls
-    }else{
-      url_fixme <- NULL
-    }
+  fixmes <- add_url_fixmes(
+    fixmes = fixmes,
+    main_url = description$get("URL"),
+    message_id = "hint_add_repo_url",
+    further_urls = description$descr$get_urls
+  )
+
+  # BugReports
+  fixmes <- add_url_fixmes(
+    fixmes = fixmes,
+    main_url = description$get("BugReports"),
+    message_id = "hint_add_bug_report_url"
+  )
+
+  fixmes_as_tibble_or_message(fixmes, where = "DESCRIPTION")
+}
+
+# fails ------------------------------------------------------------------------
+fails <- function(expr, silent = TRUE) {
+
+  inherits(try(expr, silent = silent), "try-error")
+}
 
+# add_url_fixmes ---------------------------------------------------------------
+add_url_fixmes <- function(
+  fixmes, main_url, message_id, further_urls = main_url
+) {
+
+  if (is.na(main_url)) {
+
+    fixmes <- c(fixmes, get_message(message_id))
+
+  } else if ((failing_urls <- check_urls(further_urls)) != "") {
+
+    fixmes <- c(fixmes, failing_urls)
   }
 
-  # BugReports
-  if(is.na(descr$get("BugReports"))){
-    bugreports_fixme <- "BugReports field. Indicate where to report bugs, e.g. GitHub issue tracker."
-  }else{
-    checkurls <- check_urls(descr$get("BugReports"))
-    if(checkurls != ""){
-      bugreports_fixme <- checkurls
-    }else{
-      bugreports_fixme <- NULL
-    }
+  fixmes
+}
 
+# fixmes_as_tibble_or_message ---------------------------------------------------
+fixmes_as_tibble_or_message <- function(fixmes, where, message_id = NULL) {
+
+  if (length(fixmes)) {
+
+    tibble::tibble(where = where, fixme = fixmes)
+
+  } else if (! is.null(message_id)) {
+
+    message(get_message(message_id))
   }
+}
+
+# try_to_give_opinions_readme --------------------------------------------------
+try_to_give_opinions_readme <- function(description_file) {
 
-  fixmes <- c(authors_fixme, url_fixme, bugreports_fixme)
-  fixmes <- fixmes[!is.null(fixmes)]
+  readme_path <- guess_readme(dirname(description_file))$readme_path
 
-  if(length(fixmes) >0){
+  if (is.null(readme_path)) {
```

Comment:
I'm starting to wonder whether not being able to locate a README should also be a reason for a hint. I think we can/should be confident that it means there's no README, which is not optimal?

Reply:
what do you think @hsonne? It might also be beyond the scope of this PR.

Reply:
All that I do is about code cleaning. Once this is done, I (and hopefully others) can see clearer what actually happens. I think we should keep this for later...

Reply:
👍